tripy/examples/nanogpt/weight_loader.py

```diff
@@ -114,7 +114,7 @@ def get_submodule(module, attr_name):
         if "ln" not in key:
             weight = weight.to(torch_dtype)
         param = tp.Tensor(weight.contiguous())
+        assert key in model.state_dict()
```

Comment:
We should store the `model.state_dict()` somewhere. This is an expensive call because it recurses over all child modules.